### PR TITLE
Add MonadThrow class

### LINKED ISSUE #79: MonadThrow as superclass for MonadError
I found myself needing a `MonadThrow e m`, without the need for catching, such that `throwError e *> m = throwError e`. Seems to be in a similar vein as splitting up `MonadAsk` and `MonadTell`.

### PR DESCRIPTION
**Pull request by natefaubion**
Fixes #79

### CODE REVIEW

File: src/Control/Monad/Error/Class.purs [hunk 1]
```diff
@@ -7,10 +7,22 @@ import Prelude
 import Data.Maybe (Maybe(..))
 import Data.Either (Either(..), either)
 
--- | The `MonadError` type class represents those monads which support errors via
--- | `throwError` and `catchError`.
+-- | The `MonadThrow` type class represents those monads which support errors via
+-- | `throwError`, where `throwError e` halts, yielding the error `e`.
+-- |
+-- | An implementation is provided for `ErrorT`, and for other monad transformers
```
Comment (paf31) [inline src/Control/Monad/Error/Class.purs]: This should say `ExceptT` now.

**Comment by paf31** [comment]
Are there examples which are have `throw` but not `catch`?

**Comment by natefaubion** [comment]
I don't have any common examples, no. I do internally at SlamData. Our card evaluation machinery uses `MonadThrow` to propagate errors to the user, and it doesn't make sense for us to catch these.

**Comment by natefaubion** [comment]
I'd characterize something that has `MonadThrow` but not `MonadError` as something that only supports panicking 😛

**Comment by natefaubion** [comment]
I also like to think of `Aff` as `MonadThrow`, at least internally. It does support catching exceptions, of course, but we try to always use `Either` or `ExceptT`. By only carrying a `MonadThrow` constraint, I like to think it says "Only for hard failures".

**Comment by garyb** [comment]
The error can still be caught when the running the stack, so it's not necessarily a total failure. 👍 for this!

**Pull request merged**